backend/editor/entries.py

```diff
@@ -247,33 +248,45 @@ def full_text_search(text):
     """
     Helper function used for searching a taxonomy
     """
-    normalized_text = re.sub(r"([^A-Za-z0-9])", r"\\\1", text) # Escape special characters
+    # Escape special characters
+    normalized_text = re.sub(r"([^A-Za-z0-9 _])", r"\\\1", text)
+    normalized_id_text = re.sub(r"([^A-Za-z0-9 _])", r"\\\1", normalizing(text))

```

Comment:
I don't think you need to re-normalize it ? I mean, it's fine here to match on "-"